docs/experiments-msmarco-passage.md

```diff
@@ -0,0 +1,160 @@
+# PyGaggle: Neural Ranking Baselines on [MS MARCO Passage Retrieval](https://github.com/microsoft/MSMARCO-Passage-Ranking)
+
+This page contains instructions for running various neural reranking baselines on the MS MARCO *passage* ranking task. 
+Note that there is also a separate [MS MARCO *document* ranking task](experiments-msmarco-doc.md).
+
+Prior to running this, we suggest looking at our first-stage [BM25 ranking instructions](https://github.com/castorini/anserini/blob/master/docs/experiments-msmarco-passage.md).
+We rerank the BM25 run files that contain ~ 1000 passages per query using both monoBERT and monoT5.
+
+Keeping computational resources in mind, our instructions primarily focus on a 105 query subset of the MS MARCO dev set. 
+Running these instructions with the entire MS MARCO dev set should give about the same results as that in the corresponding paper. 
+
+*Note: Run the following instructions at root of this repo. Installation must have been done from source. Make sure that you have access to a GPU*
+
+## Models
+
++ monoBERT-Large: Passage Re-ranking with BERT [(Nogueira et al., 2019)](https://arxiv.org/pdf/1901.04085.pdf)
++ monoT5-base: Document Ranking with a Pretrained Sequence-to-Sequence Model [(Nogueira et al., 2020)](https://arxiv.org/pdf/2003.06713.pdf)
+
+## Data Prep
+
+We're first going to download the queries, qrels and run files corresponding to the MS MARCO set considered. The run file is generated by following the BM25 ranking instructions. We'll store all these files in the `data` directory.
+
+```
+wget https://www.dropbox.com/s/5xa5vjbjle0c8jv/msmarco_ans_small.zip -P data
+```
+
+To confirm, `msmarco_ans_small.zip` should have MD5 checksum of `65d8007bfb2c72b5fc384738e5572f74`.
+
+Next, we extract the contents into `data`. 
+
+```
+unzip msmarco_ans_small.zip -d data
+```
+
+As a sanity check, we can evaluate the first-stage retrieved documents using the official MS MARCO evaluation script.
+
+```
+python evaluate/msmarco/msmarco_eval.py data/msmarco_ans_small/qrels.dev.small.tsv data/msmarco_ans_small/run.dev.small.tsv
+```
+
+The output should be:
+
+```
+#####################
+MRR @10: 0.15906651549508694
+QueriesRanked: 105
+#####################
+```
+
+Let's download and extract the pre-built MS MARCO index into `indexes`:
+
+```
+wget https://git.uwaterloo.ca/jimmylin/anserini-indexes/raw/master/index-msmarco-passage-20191117-0ed488.tar.gz -P indexes
+tar xvfz indexes/index-msmarco-passage-20191117-0ed488.tar.gz -C indexes
+```
+
+## Model Prep
+
+Let's download and extract monoBERT into `models`:
+
+```
+wget https://www.dropbox.com/s/jr0hpksboh7pa48/monobert_msmarco_large.zip -P models
```

Comment:
Perhaps we should use BERT-base so it is a fairer comparison with T5-base?

Reply:
I think T5-base is situated in between BERT-base and BERT-large in terms of model complexity right? Based on that, I feel like we should just go with T5-base. We can point out that on running the entire set T5-base will perform better than BERT-base (just randomness in this subset).

Reply:
SGTM!